_websiteutils/make_redirects_links.py

```diff
@@ -145,16 +140,12 @@ def do_canonicals(dname):
     to the newest version.
     """
     _log.debug(f"Walking {dname}")
-    for root, dirs, files in os.walk(dname):
-        for name in files:
-            fullname = os.path.join(root, name)
-            p = pathlib.Path(fullname)
-            _log.debug(f"Checking {fullname}")
-            if name.endswith((".htm", ".html")):
-                basename = pathlib.Path(*p.parts[1:])
-                last = findlast(basename, tocheck)
-                if last is not None:
-                    update_canonical(fullname, last, dname == tocheck[1])
+    for fullname in dname.rglob("*.html"):
```

Comment:
Note, `Path.rglob` doesn't seem to support multiple patterns, but we do not have any `.htm` files.